src/Development/IDE/Plugin/CodeAction/Rules.hs

```diff
@@ -0,0 +1,59 @@
+module Development.IDE.Plugin.CodeAction.Rules
+  ( rulePackageExports,
+    getVisiblePackageExports
+  )
+where
+
+import Data.HashMap.Strict (fromListWith)
+import qualified Data.HashMap.Strict as Map
+import Data.Text ( Text, pack,)
+import Data.Traversable (forM)
+import Development.IDE.Core.RuleTypes
+import Development.IDE.Core.Shake
+import Development.IDE.GHC.Util
+import Development.IDE.Plugin.CodeAction.RuleTypes
+import Development.IDE.Types.Location
+import Development.Shake
+import HscTypes (hsc_dflags, IfaceExport, mi_exports)
+import LoadIface
+import Maybes
+import Module (Module (..), ModuleName, moduleNameString)
+import Packages (explicitPackages, exposedModules, packageConfigId)
+import TcRnMonad (WhereFrom (ImportByUser), initIfaceLoad)
+import GHC (DynFlags(pkgState))
+import Control.Monad.Trans (MonadTrans(lift))
+
+getVisiblePackageExports :: NormalizedFilePath -> Action [PackageExportsMap]
+getVisiblePackageExports f = fmap (fromMaybe []) $ runMaybeT $ do
+        session <- MaybeT $ use GhcSession f
+        let exposedPkgs = explicitPackages $ pkgState (hsc_dflags $ hscEnv session)
+        traverse (\p -> lift $ use_ (fromUnitId p) f) exposedPkgs
+
+rulePackageExports :: Rules ()
+rulePackageExports = define $ \p file -> do
```

Comment:
A type annotation on `p` would be quite useful to see which rule is being defined here.